x-pack/solutions/observability/plugins/apm/common/service_map/group_resource_nodes.ts

```diff
@@ -49,7 +51,16 @@ function groupConnections({
 
   const groups = new Map<string, { id: string; sources: string[]; targets: string[] }>();
   for (const [target, sources] of sourcesByTarget) {
-    const groupId = `resourceGroup{${[...sources].sort().join(';')}}`;
+    // Get spanSubtype from the target node
+    const targetNode = nodesMap.get(target);
+    const targetData = targetNode && isNode(targetNode) ? targetNode.data : undefined;
+    const spanSubtype = targetData?.[SPAN_SUBTYPE];
+
+    // Group by source services AND spanSubtype
+    const groupId = spanSubtype
+      ? `resourceGroup{${[...sources].sort().join(';')}|${spanSubtype}}`
+      : `resourceGroup{${[...sources].sort().join(';')}}`;
```

Comment:
I'm a bit concerned about grouping by `spanSubtype` after checking the data on the edge cluster


<img width="776" height="689" alt="image" src="https://github.com/user-attachments/assets/3f732f04-5cf3-490d-9bcc-7efdfae60747" />


I suspect doing this will create on group per resource

Reply:
Tested it locally, and it did create one group per resource

<img width="800" height="683" alt="image" src="https://github.com/user-attachments/assets/253a9d15-44d3-43c8-82ed-32e09a172df6" />


<img width="800" height="673" alt="image" src="https://github.com/user-attachments/assets/86f03974-520a-4d64-aff5-4e81f10b2482" />

Reply:
Fixed, rolled back grouping by spanSubType